src/IIIFPresentation/Repository/PresentationContext.cs

```diff
@@ -30,9 +35,43 @@ protected override void OnModelCreating(ModelBuilder modelBuilder)
         modelBuilder.Entity<Collection>(entity =>
         {
             entity.HasKey(e => new {e.Id, e.CustomerId});
-            entity.HasIndex(e => new { e.CustomerId, e.Slug, e.Parent }).IsUnique();
             
             entity.Property(e => e.Label).HasColumnType("jsonb");
+            
+            // TODO: is there issues on deletions for hierarchy with manifest/collections with the same key?
+            entity.HasMany(e => e.Hierarchy)
+                .WithOne(e => e.Collection)
+                .HasForeignKey(e => new { e.CollectionId, e.CustomerId })
+                .HasPrincipalKey(e => new { e.Id, e.CustomerId })
+                .OnDelete(DeleteBehavior.Cascade);
+        });
+        
+        modelBuilder.Entity<Manifest>(entity =>
+        {
+            entity.HasKey(e => new {e.Id, e.CustomerId});
+            
+            entity.HasMany(e => e.Hierarchy)
+                .WithOne(e => e.Manifest)
+                .HasForeignKey(e => new { e.ManifestId, e.CustomerId })
+                .HasPrincipalKey(e => new { e.Id, e.CustomerId })
+                .OnDelete(DeleteBehavior.Cascade);
+        });
+        
+        modelBuilder.Entity<Hierarchy>(entity =>
```

Comment:
currently the API assumes that there will always be a canonical path on retrieval, but this isn't enforced by the database - I did leave this out as it might complicate updates, but is this worth adding?

Reply:
decision is to default `canonical` to `true` for now

Reply:
For future reference - ticket created to handle non-canonical paths. This isn't required initially.